barretenberg/cpp/src/barretenberg/ultra_honk/decider_proving_key.cpp

```diff
@@ -30,6 +30,146 @@ template <IsUltraFlavor Flavor> size_t DeciderProvingKey_<Flavor>::compute_dyadi
+        } else {
+            selector = Polynomial(block.get_fixed_size(is_structured), proving_key.circuit_size, block.trace_offset);
+        }
+    }
+
+    // Set the other non-gate selector polynomials (e.g. q_l, q_r, q_m etc.) to full size
+    for (auto& selector : proving_key.polynomials.get_non_gate_selectors()) {
+        selector = Polynomial(proving_key.circuit_size);
+    }
+}
+
+template <IsUltraFlavor Flavor>
+void DeciderProvingKey_<Flavor>::allocate_table_lookup_polynomials(const Circuit& circuit)
+{
+    PROFILE_THIS_NAME("allocate_table_lookup_polynomials");
+
+    const size_t max_tables_size = std::min(static_cast<size_t>(MAX_LOOKUP_TABLES_SIZE), dyadic_circuit_size - 1);
+    size_t table_offset = dyadic_circuit_size - max_tables_size;
+    ASSERT(dyadic_circuit_size > max_tables_size);
+
+    // Allocate the polynomials containing the actual table data
+    if constexpr (IsUltraFlavor<Flavor>) {
+        for (auto& poly : proving_key.polynomials.get_tables()) {
+            poly = Polynomial(max_tables_size, dyadic_circuit_size, table_offset);
+        }
+    }
+
+    // Allocate the read counts and tags polynomials
+    proving_key.polynomials.lookup_read_counts = Polynomial(max_tables_size, dyadic_circuit_size, table_offset);
+    proving_key.polynomials.lookup_read_tags = Polynomial(max_tables_size, dyadic_circuit_size, table_offset);
+    ZoneScopedN("allocating lookup and databus inverses");
+
+    // Allocate the log derivative lookup argument inverse polynomial
+    const size_t lookup_offset = static_cast<size_t>(circuit.blocks.lookup.trace_offset);
+    const size_t lookup_inverses_start = std::min(lookup_offset, table_offset);
+    const size_t lookup_inverses_end =
+        std::min(dyadic_circuit_size,
+                 std::max(lookup_offset + circuit.blocks.lookup.get_fixed_size(is_structured),
+                          table_offset + MAX_LOOKUP_TABLES_SIZE));
+    proving_key.polynomials.lookup_inverses =
+        Polynomial(lookup_inverses_end - lookup_inverses_start, dyadic_circuit_size, lookup_inverses_start);
+}
+
+template <IsUltraFlavor Flavor>
+void DeciderProvingKey_<Flavor>::allocate_ecc_op_polynomials(const Circuit& circuit)
+    requires IsMegaFlavor<Flavor>
+{
+    PROFILE_THIS_NAME("allocate_ecc_op_polynomials");
+
+    // Allocate the ecc op wires and selector
+    const size_t ecc_op_block_size = circuit.blocks.ecc_op.get_fixed_size(is_structured);
+    const size_t op_wire_offset = circuit.blocks.ecc_op.trace_offset;
+    for (auto& wire : proving_key.polynomials.get_ecc_op_wires()) {
+        wire = Polynomial(ecc_op_block_size, proving_key.circuit_size, op_wire_offset);
+    }
+    proving_key.polynomials.lagrange_ecc_op = Polynomial(ecc_op_block_size, proving_key.circuit_size, op_wire_offset);
+}
+
+template <IsUltraFlavor Flavor>
+void DeciderProvingKey_<Flavor>::allocate_databus_polynomials(const Circuit& circuit)
```

Comment:
can you add a `PROFILE_THIS_NAME` tag on this function as well

Reply:
yes! but will do in a follow on because I'm going to do lot of cleanup of the profiling/labels